app/src/main/java/org/mozilla/fenix/wallpapers/WallpaperDownloader.kt

```diff
@@ -73,9 +73,12 @@ class WallpaperDownloader(
     private fun Wallpaper.Remote.toMetadata(context: Context): List<WallpaperMetadata> =
         listOf("landscape", "portrait").flatMap { orientation ->
             listOf("light", "dark").map { theme ->
-                val remoteParent = this::class.simpleName!!.lowercase()
                 val localPath = "wallpapers/$orientation/$theme/$name.png"
-                val remotePath = "${context.resolutionSegment()}/$orientation/$theme/$remoteParent/$name.png"
+                val remotePath = "${context.resolutionSegment()}/" +
```

Comment:
Simple unit tests for this type of code can go a long way to avoid regressions in the future since content is dynamically downloaded. I'll leave that to you to file as a follow-up when time permits.

Reply:
I completely agree! The rate the requirements have changed has made it difficult to write this feature at the level of quality that I would normally expect of myself. Much of this code also clearly depends on platform types, which would mean structuring our dependencies in a more testable fashion. A more robust test suite is something I would like to include as part of [uplifting this feature to A-C](https://github.com/mozilla-mobile/android-components/issues/11679)